ctlevent.js

```diff
@@ -4,11 +4,28 @@
 var propertiesString = function(properties) {
     var propString = '';
     for (var i in properties) {
-        propString += '<span>' + i + ': </span>' + properties[i] + '</br>';
+        propString += '<span class="ctl-property-name">' + properties[i].name + ': </span>';
+        for (var j in properties[i].values) {
+            propString += '<span class="ctl-property-value">' + properties[i].values[j] + '</span> ';
+        }
+        propString += '</br>';
     }
     return propString;
 };
 
+var addProperty = function(that, propList) {
+    function checkPropertyName(element) {
+        return element.name == propList[0];
+    }
+
+    var found;
+    if ((found = that.propertyArray.findIndex(checkPropertyName)) > -1) {
+        that.propertyArray[found].values.push(propList[1]);
```

Comment:
This is a great re-factoring, and I'm going to make some updates to this branch.

Some feedback:
* `findIndex` is an ES6 feature, just like the backtick template syntax. So it works on current browsers, but it's brand new and in the past we've used underscore/lodash for array utilities like this: https://lodash.com/docs/4.17.4#findIndex
* Because `addProperty` is modifying the CTLEvent instance, it should be an instance method on this class.